scripts/template_matcher.py

```diff
@@ -19,35 +31,48 @@ def __init__ (self, images, min_match_count=10, good_thresh=0.7):
         # for potential tweaking
         self.min_match_count = min_match_count
         self.good_thresh = good_thresh #use for keypoint threshold
+        self.matcher = cv2.BFMatcher()
 
-        #TODO: precompute keypoints for template images
+        self.get_template_keypoints(images)
 
+    def get_template_keypoints(self, template_imgs):
```

Comment:
I like how you wrapped this code in a function!  I didn't do that in my implementation, but I wish I did. It makes the all the code in __init__ to be on the same abstraction level.